packages/amplify-util-uibuilder/src/commands/utils/syncAmplifyUiBuilderComponents.ts

```diff
@@ -71,3 +74,47 @@ export const generateUiBuilderThemes = (context: $TSContext, themeSchemas: any[]
   );
   return themeResults;
 };
+
+/**
+ * Returns instances of StudioForm from form schemas
+ */
+export const generateUiBuilderForms = (
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  context: $TSContext,
+  formSchemas: any[],
+  dataSchema?: GenericDataSchema,
+  autoGenerateForms?: boolean,
+): CodegenResponse<StudioForm>[] => {
+  const modelMap: { [model: string]: Set<'create' | 'update'> } = {};
+  if (dataSchema?.dataSourceType === 'DataStore' && autoGenerateForms) {
+    Object.entries(dataSchema.models).forEach(([name, model]) => {
+      if(checkIsSupportedAsForm(model) && !model.isJoinTable) {
+        modelMap[name] = new Set(['create', 'update']);
+      }
+    });
+  }
+  const codegenForm = (schema: StudioForm): CodegenResponse<StudioForm> => {
+    try {
+      const form = createUiBuilderForm(context, schema, dataSchema);
+      return { resultType: 'SUCCESS', schema: form };
+    } catch (e) {
+      printer.debug(`Failure caught processing ${schema.name}`);
+      printer.debug(e);
+      return { resultType: 'FAILURE', schemaName: schema.name, error: e };
+    }
+  };
+  const formResults = formSchemas.map((schema: StudioForm) => {
+    if (schema?.dataType && schema.dataType?.dataSourceType === 'DataStore') {
+      modelMap[schema.dataType.dataTypeName]?.delete(schema.formActionType);
+    }
+    return codegenForm(schema);
+  });
+
+  // append remaining models
+  formResults.push(...generateBaseForms(modelMap).map(codegenForm));
+
+  printer.debug(
```

Comment:
I don't recall noticing `debug` being used as a method to trace log entries, is this something that was discussed or we should use the AmplifyLogger?

Reply:
we use debug when printing output happy to add the output to a logger as well